server/controllers/finance/purchases.js

```diff
@@ -361,58 +362,82 @@ function find(options) {
  * if it is completed or partially entered
  */
 function purchaseStatus(req, res, next) {
-  const status = {};
   const FROM_PURCHASE_ID = 1;
+  const PURCHASE_PARTIALLY_RECEIVED = 4;
+  const PURCHASE_FULLY_RECEIVED = 3;
+  const PURCHASE_EXCESSIVELY_RECEIVED = 6;
+
+  const glb = {};
   const purchaseUuid = db.bid(req.params.uuid);
-  const sql = `
-    SELECT IFNULL(SUM(m.quantity * m.unit_cost), 0) AS movement_cost, p.cost, CEIL(DATEDIFF(m.date, p.date)) AS delay
+
+  const sqlQuantities = `
+    SELECT SUM(t.purchase_quantity) AS purchase_quantity, SUM(t.movement_quantity) AS movement_quantity
+    FROM (
+      (
+        SELECT IFNULL(SUM(pi.quantity), 0) AS purchase_quantity, 0 AS movement_quantity FROM purchase p
+        JOIN purchase_item pi ON pi.purchase_uuid = p.uuid
+        WHERE p.uuid = ?
+      ) UNION (
```

Comment:
Just to be safe, this should probably be `UNION ALL`.  It might remove (duplicate) rows from the plain `UNION` statement.